common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -833,6 +834,50 @@ public static int hashCodeAscii(CharSequence bytes) {
         return null;
     }
 
+    private static final class Mpsc {
+        private static final boolean USE_MPSC_CHUNKED_ARRAY_QUEUE;
+
+        private Mpsc() {
+        }
+
+        static {
+            Object unsafe = null;
+            if (hasUnsafe()) {
+                // jctools goes through its own process of initializing unsafe; of
+                // course, this requires permissions which might not be granted to calling code, so we
+                // must mark this block as privileged too
+                unsafe = AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                    @Override
+                    public Object run() {
+                        // force JCTools to initialize unsafe
+                        return UnsafeAccess.UNSAFE;
+                    }
+                });
+            }
+
+            if (unsafe == null) {
+                logger.debug("org.jctools-core.MpscChunkedArrayQueue: unavailable");
+                USE_MPSC_CHUNKED_ARRAY_QUEUE = false;
+            } else {
+                logger.debug("org.jctools-core.MpscChunkedArrayQueue: available");
+                USE_MPSC_CHUNKED_ARRAY_QUEUE = true;
+            }
+        }
+
+        private static <T> Queue<T> newMpscQueue(final int maxCapacity) {
```

Comment:
I know... Still the method should be package private imho

> Am 06.08.2016 um 21:28 schrieb Jason Tedor <notifications@github.com>:
> 
> In common/src/main/java/io/netty/util/internal/PlatformDependent.java:
> 
> > +                        // force JCTools to initialize unsafe
> > +                        return UnsafeAccess.UNSAFE;
> > +                    }
> > +                });
> > +            }
> > +
> > +            if (unsafe == null) {
> > +                logger.debug("org.jctools-core.MpscChunkedArrayQueue: unavailable");
> > +                USE_MPSC_CHUNKED_ARRAY_QUEUE = false;
> > +            } else {
> > +                logger.debug("org.jctools-core.MpscChunkedArrayQueue: available");
> > +                USE_MPSC_CHUNKED_ARRAY_QUEUE = true;
> > +            }
> > +        }
> > +
> > +        private static <T> Queue<T> newMpscQueue(final int maxCapacity) {
> The public API of PlatformDependent has not changed, the public PlatformDependent#newMpscQueue and PlatformDependent#newMpscQueue(int) methods are still present, they just chain through to this private method (which would be inaccessible even if public since the static nested class Mpsc is private here too).
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>